PusherClient/EventEmitter.cs

```diff
@@ -16,7 +16,9 @@ public void Bind(string eventName, Action<dynamic> listener)
         {
             if(_eventListeners.ContainsKey(eventName))
             {
-                _eventListeners[eventName].Add(listener);
+                // To avoid triggering the same bind multiple times

```

Comment:
About the EventEmitter, I got it. I am not familiar with this approach, but, it makes sense.

About the bindings and your example, how much you will wait for the connection? 
It is undetermined, right? 

I think thats the reason why the example use the events to subscribe to the channel and bind the methods. 

How you will do the "wait for the connection"? A loop testing the status?